### PR TITLE
task done

### PR DESCRIPTION
**Pull request by Manankin**
[DEMO LINK](https://Manankin.github.io/react_people-table-basics/)

### CODE REVIEW

**Review by VasiliyYarosh — COMMENTED** [review]
Good Job. Have some advice for u. Fix the demo link. Delete comments which are not necessary. 
image

**Review by MaxKalalashnyk02 — APPROVED** [review]
Good work!

File: src/PeopleContext.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React, { createContext, useState } from 'react';
+import { Person } from './types';
+
+type ContextProps = {
```
Comment (DanilWeda) [inline src/PeopleContext.tsx:4]: use interface prefer. 
Types have unique features but the interface is best for scaling.

File: src/PeopleContext.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  peopleList: Person[];
+  setPeopleList: (list: Person[]) => void;
+  isLoading: boolean;
+  setIsLoading: (b: boolean) => void;
+  isError: boolean;
+  setIsError: (b: boolean) => void;
+};
+
+type Props = {
```
Comment (DanilWeda) [inline src/PeopleContext.tsx:13]: use interface prefer. 
Types have unique features but the interface is best for scaling.

File: src/pages/People.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  useEffect(() => {
+    setIsLoading(true);
+    setIsError(false);
+
+    getPeople()
+      .then(setPeopleList)
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
```
Comment (DanilWeda) [inline src/pages/People.tsx:22]: you can disabled eslint rule  in this case

File: src/components/personLink/PersonLink.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import { useContext } from 'react';
+import { useParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { Person } from '../../types';
+import { PeopleList } from '../../PeopleContext';
+
+type Props = {
```
Comment (DanilWeda) [inline src/components/personLink/PersonLink.tsx:7]: use interface prefer. 
Types have unique features but the interface is best for scaling.

File: src/components/personLink/PersonLink.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  person: Person;
+};
+
+export const PersonLink: React.FC<Props> = ({ person }) => {
+  const { peopleList } = useContext(PeopleList);
+  const { slug } = useParams();
+
+  const mother = peopleList.find(m => m.name === person.motherName);
+  const father = peopleList.find(f => f.name === person.fatherName);
+  // const selectedPerson = person.slug === slug;
```
Comment (DanilWeda) [inline src/components/personLink/PersonLink.tsx:17]: remove commented

**Review by DanilWeda — CHANGES_REQUESTED** [review]
Good job! 
Please, redeploy the app.
The demo does not work. 
Thanks!

File: src/components/peopleTable/PeopleTable.tsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+          <th>Father</th>
+        </tr>
+      </thead>
+
+      <tbody>
+        {peopleList.map(person => (
+          <PersonLink person={person} key={person.slug} />
+        ))}
+
+        {/* <tr data-cy="person">
```
Comment (olya-shyian) [inline src/components/peopleTable/PeopleTable.tsx:29]: please remove commented code

File: src/pages/People.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+            <p data-cy="noPeopleMessage">
+              There are no people on the server
+            </p>
+          )}
+
+          {!!peopleList.length && !isLoading && (
+            <PeopleTable />
+          )}
+
+        </div>
```
Comment (olya-shyian) [inline src/pages/People.tsx:50]: empty line

File: src/Root.tsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import {
+  HashRouter, Navigate, Route, Routes,
+} from 'react-router-dom';
+import { App } from './App';
+import { Home } from './pages/Home';
+import { People } from './pages/People';
+import { NotFound } from './pages/NotFound';
+
+export const Root = () => {
+  return (
```
Comment (olya-shyian) [inline src/Root.tsx:10]: in such case please return jsx without using word `return`, in round brackets

File: src/App.tsx [hunk 9]
```diff
@@ -12,161 +17,26 @@ export const App = () => (
...
+          <NavLink
+            to="/people"
+            className={getLinkClass}
           >
             People
-          </a>
+          </NavLink>
         </div>
       </div>
     </nav>
```
Comment (olya-shyian) [inline src/App.tsx:35]: it would be great to create a component for navigation

**Review by olya-shyian — CHANGES_REQUESTED** [review]
Great!👍

**Review by olya-shyian — APPROVED** [review]
Thank you for the updates!☘️

**Pull request closed without merging**